src/components/contextual/pages/pool/PoolTransactionsCard/BoostedPoolActivities/Table.vue

```diff
@@ -0,0 +1,270 @@
+<script setup lang="ts">
+import { defineEmits, defineProps, computed } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { formatDistanceToNow } from 'date-fns';
+import { groupBy } from 'lodash';
+
+import useWeb3 from '@/services/web3/useWeb3';
+import { FullPool, PoolSwap } from '@/services/balancer/subgraph/types';
+
+import useTokens from '@/composables/useTokens';
+import useNumbers from '@/composables/useNumbers';
+import useBreakpoints from '@/composables/useBreakpoints';
+
+import { ColumnDefinition } from '@/components/_global/BalTable/BalTable.vue';
+
+import { bnum } from '@/lib/utils';
+
+/**
+ * TYPES
+ */
+type TokenAmount = {
+  address: string;
+  amount: string;
+};
+
+type SwapRow = {
+  label: string;
+  timestamp: number;
+  formattedDate: string;
+  value: number;
+  formattedValue: string;
+  isInvest: boolean;
+  tx: string;
+  tokenAmounts: TokenAmount[];
+};
+
+type Props = {
+  tokens: string[];
+  poolSwaps: PoolSwap[];
+  isLoading?: boolean;
+  isLoadingMore?: boolean;
+  loadMore?: () => void;
+  isPaginated?: boolean;
+  noResultsLabel?: string;
+  poolAddress: string;
+  pool: FullPool;
+};
+
+/**
+ * PROPS & EMITS
+ */
+const props = withDefaults(defineProps<Props>(), {
+  isLoading: false,
+  isLoadingMore: false,
+  isPaginated: false
+});
+
+const emit = defineEmits(['loadMore']);
+
+/**
+ * COMPOSABLES
+ */
+const { fNum } = useNumbers();
+const { t } = useI18n();
+const { priceFor } = useTokens();
+const { upToLargeBreakpoint } = useBreakpoints();
+const { explorerLinks } = useWeb3();
+
+/**
+ * COMPUTED
+ */
+const columns = computed<ColumnDefinition<SwapRow>[]>(() => [
+  {
+    name: t('action'),
+    id: 'action',
+    accessor: 'tx',
+    Cell: 'actionCell',
+    width: 150,
+    sortable: false
+  },
+  {
+    name: t('details'),
+    id: 'details',
+    accessor: '',
+    Cell: 'detailsCell',
+    width: 325,
+    sortable: false
+  },
+  {
+    name: t('value'),
+    id: 'value',
+    accessor: 'value',
+    Cell: 'valueCell',
+    align: 'right',
+    className: 'align-center w-40',
+    sortKey: pool => pool.value,
+    width: 125
+  },
+  {
+    name: t('time'),
+    id: 'timeAgo',
+    accessor: 'timestamp',
+    Cell: 'timeCell',
+    align: 'right',
+    sortKey: pool => pool.timestamp,
+    width: 200
+  }
+]);
+
+const swapRows = computed<SwapRow[]>(() => {
+  if (props.isLoading) {
+    return [];
+  }
+
+  const groupedSwaps = Object.entries(groupBy(props.poolSwaps, 'tx'));
+
+  return groupedSwaps.map(([tx, swaps]) => {
+    const { tokenOut, timestamp } = swaps[0];
+
+    const isInvest = tokenOut === props.pool.address;
```

Comment:
This logic might change, I believe its correct, but needs to be audited (this assumes that all swaps are only investment/withdraw ones)